### ISSUE TITLE
Control model parameters for separate system component

### ISSUE THREAD

**Issue by pckroon** [marrink-lab/vermouth-martinize#349]
Let's say you have a system consisting of a polymer, a protein, and some DNA and RNA. Mix in some lipids for good measure. Generating the "basic" topology for this system should work, given that you have all the required Blocks and Links in place. 

However, you have only very coarse control over the elastic network (or any other dress up done by separate processors).

You need to have the option to apply the (e.g.) elastic network processor on separate parts of the system with different parameters.
- From where do you need to be able to control this? I'd say data files for defaults and CLI for overrides.
- This should be applicable to *all* processors. You could say you don't want to run repair graph on the lipids, since you know those are complete and well behaved.
- How to implement this? This steers towards file-defined workflows, where the CLI is only for overriding the defaults. Martinize2 will then assemble and execute the pipeline dynamically.

**Comment by jbarnoud** [comment]
That was more or less the point of spoon that I never got to write: you define your workflow as something like a yaml file by listing what processors to run and how to run them. This way you are not fighting against the CLI, instead you build a template for your project and can adapt it system per system.

**Comment by pckroon** [comment]
Yes, but. Most users will not (want to) write extensive yaml (toml?) files, these should come with the force fields we provide. Here's a few random thoughts:
- The workflow will be a DAG.
- The output ff should determine the processors to run from universal to output, and the input ff should determine the processors to run from input to universal. 
- The resulting workflow from input to output determines the available CLI flags and their defaults. This will probably also require us to clean up/define the Processor API more strictly. 
- It must be possible to add optional processors to workflow, that will be hidden behind a CLI flag (such as EN is now).
- We probably want to build upon an existing workflow manager/library/whatever? E.g. https://github.com/spotify/luigi.
- Workflow steps could be other workflows. The library may provide some common (partial) workflows, such as "repair the graph and make sure all modifications are flagged". This will also help unifying the CLI flags.
- It should be possible to run a processor multiple times on separate parts of the system (e.g. once for protein, once for DNA). This will require some form of selection language ('is protein', 'contains protein', ...). Selections should be per molecule, residue or atom. The trickiest thing will be dealing with overlapping selections. For example: run the elastic network processor on 'molecule contains protein', and on 'molecule contains DNA'. If this is run on a protein-DNA hybrid, which one determines the final parameters?
